### PR TITLE
feat: Navigate authenticated users to /home and initiate homepage design

### PR DESCRIPTION
**Pull request by adelhelj**
This commit introduces a new feature where users authenticated via Firebase are now redirected to the /home route, aligning with our user flow requirements. Additionally, it lays the groundwork for the homepage design, which is currently in a work-in-progress state. The design follows our initial Figma prototype and will be refined in subsequent updates. Authenticated users can expect a seamless transition to a the user home page

I took an initial stab tonight at the home page, so this is just a draft but I wanted to get some work done since we're approaching week 5. 

To get the React app running, cd into the frontend directory. From there you must install 3 things:
npm install
npm install firebase
npm install react-router-dom

From there 'npm start'

TLDR:
When the user is authenticated on the splash page, they will be directed to the user homepage (CSS styling is WIP - aims to match Figma proto). There is also a logout button that redirects the user to the splash page.

### CODE REVIEW

File: frontend/src/components/SplashPage.js [hunk 1]
```diff
@@ -1,32 +1,21 @@
...
     const provider = new GoogleAuthProvider();
     signInWithPopup(auth, provider)
       .then((result) => {
-        // This gives you a Google Access Token. You can use it to access the Google API.
-        // const credential = GoogleAuthProvider.credentialFromResult(result);
-        // const token = credential.accessToken;
-        // The signed-in user info.
-        // const user = result.user;
-        // ...
+        // After sign-in, navigate to the UserHomepage
```
Comment (GPropersi) [inline frontend/src/components/SplashPage.js:15]: HTTP POST request to backend goes here - first to auth, then register if they are new.

If register, then form for username should pop up asking them to choose a username

File: frontend/src/components/SplashPage.js [hunk 2]
```diff
@@ -1,32 +1,21 @@
...
+        navigate('/home');
       }).catch((error) => {
         // Handle Errors here.
-        // const errorCode = error.code;
```
Comment (GPropersi) [inline frontend/src/components/SplashPage.js:23]: HTTP Errors should be handled here and displayed to the user

File: frontend/src/components/SplashPage.js [hunk 3]
```diff
@@ -35,45 +24,30 @@ const SplashPage = () => {
     const provider = new GithubAuthProvider();
     signInWithPopup(auth, provider)
       .then((result) => {
-        // This gives you a GitHub Access Token. You can use it to access the GitHub API.
```
Comment (GPropersi) [inline frontend/src/components/SplashPage.js:38]: Same as above, HTTP POST request to backend goes here... one flow should cover both of these authentication methods

File: frontend/src/components/SplashPage.js [hunk 4]
```diff
@@ -35,45 +24,30 @@ const SplashPage = () => {
...
-        // The signed-in user info.
-        // const user = result.user;
-        // ...
-      }).catch((error) => {
+        // After sign-in, navigate to the UserHomepage
+        navigate('/home');
+      })
+      .catch((error) => {
         // Handle Errors here.
-        // const errorCode = error.code;
```
Comment (GPropersi) [inline frontend/src/components/SplashPage.js:46]: Same as above, HTTP Errors should be handled here and displayed to the user

**Review by GPropersi — APPROVED** [review]
Left a few notes where the HTTP requests should go for user authentication or registration. Looks good otherwise, nice job!

File: frontend/src/components/SplashPage.js (hunk 1 above)
Comment (adelhelj) [reply frontend/src/components/SplashPage.js:15]: Awesome, thank you Giovanni. Really excited to start incorporating these towards an initial prototype.

**Review by barreryu — DISMISSED** [review]

**Review by GPropersi — APPROVED** [review]
Nice feature to see changes since last review! Thanks GitHub

**Review by barreryu — APPROVED** [review]
